src/Speck.js

```diff
@@ -39,12 +39,23 @@ class Speck {
     this._validate();
   }
 
+  __initFieldvalue(field, data) {
+    const hasValue = !isNil(data[field]);
```

Comment:
create a function for `this.schema[field].defaultValue` would improve the readbility, ;)